readme.txt
=== Very Simple Event List ===
Contributors: Guido07111975
Donate link: https://www.paypal.com/cgi-bin/webscr?cmd=_donations&business=donation%40guidovanderleest%2enl
Version: 6.1
License: GNU General Public License v3 or later
License URI: http://www.gnu.org/licenses/gpl-3.0.html
Requires at least: 4.6
Tested up to: 4.7
Stable tag: trunk
Tags: simple, upcoming, past, current, event, list, manager, widget


This is a very simple plugin to display a list of events. Use a shortcode to display events on a page or use the widget.


== DESCRIPTION ==
= About =
This is a very simple plugin to display a list of events. 

* Use shortcode `[vsel]` to display upcoming events (including today)
* Use shortcode `[vsel-past-events]` to display past events
* Use shortcode `[vsel-current-events]` to display current events
* Use shortcode `[vsel-all-events]` to display all events

You can also list events in your sidebar using the widget.

For more info please take a look at the Installation section.

= Question? = 
Please take a look at the Installation and FAQ section.

= Translation =
Not included but plugin supports WordPress language packs.

More [translations](https://translate.wordpress.org/projects/wp-plugins/very-simple-event-list) are very welcome!

= Credits =
Without the WordPress codex and help from the WordPress community I was not able to develop this plugin, so: thank you!

Enjoy!


== INSTALLATION == 
= How to use =
After installation go to Events and start adding your events.

* Use shortcode `[vsel]` to display upcoming events (including today)
* Use shortcode `[vsel-past-events]` to display past events
* Use shortcode `[vsel-current-events]` to display current events
* Use shortcode `[vsel-all-events]` to display all events

= Widget =
You can also list events in your sidebar using the widget.

The pagination is hidden because it's not working properly in a widget. 

But you can set a link to a page with all events.

= Settingspage =
Via Settings > VSEL you can:

* Hide date
* Show a summary instead of all content
* Link title to the event page
* Hide elements in event list
* Keep events and settings when uninstalling plugin

= Shortcode attributes =
Events from certain categories: `[vsel event_cat="first-category, second-category"]`

You should enter the category slug (a slug is not always the same as category name).

Set amount of events per page: `[vsel posts_per_page=5]` 

This will overwrite amount set in Settings > Reading.

Change default events order: `[vsel order=desc]`

= Change labels =
* Change date label: `[vsel date_label="Event date: %s"]`
* Change start date label: `[vsel start_label="Event start date: %s"]`
* Change end date label: `[vsel end_label="Event end date: %s"]`
* Change time label: `[vsel time_label="Event time: %s"]` 
* Change location label: `[vsel location_label="Event location: %s"]` 
* Hide a label: `[vsel date_label="%s"]`

Multiple attributes: `[vsel posts_per_page=5 date_label="Event date: %s"]` 

= Widget attributes =
The widget supports the same attributes.

= Examples =
* Events from certain categories: `event_cat="first-category, second-category"`
* Set amount of events per page: `posts_per_page=5`
* Change default events order: `order=desc`
* Change date label: `date_label="Event date: %s"`
* Hide a label: `date_label="%s"`
* Multiple attributes: `posts_per_page=5 date_label="Event date: %s"` 

= Single event =
Single event display is supported and template file single.php is used for that. This file is located in your theme folder.

= Event category page =
It will only display properly when using the shortcode on a page.

Example: `[vsel event_cat="first-category, second-category"]`

= Uninstall =
If you uninstall plugin via WP dashboard all events and settings are removed from database.

It deletes all posts with (custom) post type 'event'.

You can avoid this via Settings > VSEL.


== Frequently Asked Questions ==
= How can I change date format? =
You can set date format in WP dashboard via Settings > General.

The datepicker and date input field in backend accept 2 numeric date formats: day-month-year (30-01-2016) and year-month-day (2016-01-30).

If the date format set in WP dashboard is not numeric it will be changed into 1 of these 2 numeric date formats.  

= How do I set plugin language? =
Plugin uses the WP Dashboard language, set in Settings > General.

If plugin language pack is not available, language fallback will be English.

= How do I style the event list? =
It mostly depends on the stylesheet of your theme.

You can change style (CSS) using for example the [Very Simple Custom Style](https://wordpress.org/plugins/very-simple-custom-style) plugin.

For non CSS related customizations of the single event display you can add a file called single-event.php in your theme folder and customize this file to your needs.

= What do you mean with current events? =
Current events are events I can visit today. So this can be an one-day or multi-day event.

= Are events also listed on time? =
No, because input field for time is a regular text input this is not possible.
 
= How can I set number of events on a page? =
You can find more info about this at the Installation section.

= Can I display a summary instead of all content? =
You can find more info about this at the Installation section.

= Can I change the summary lenght? =
Yes, you can set a custom summary while adding an event. This will replace the default summary.

= Can I display certain events only? =
You can find more info about this at the Installation section.

= Is it possible to display events in a widget? =
You can find more info about this at the Installation section.

= Why is a single event not displayed properly? =
You can find more info about this at the Installation section.

= Why is the event category page not displayed properly? =
You can find more info about this at the Installation section.

= Can I hide the event labels? =
You can find more info about this at the Installation section.

= What do you mean with link and link label? =
It's a link to another page or website for more info. 

You can enter the full URL or without the http part (the http part will be auto added if not entered).

= Can I use multiple shortcodes on the same page? =
Yes, as far as I know there are no issues when using multiple shortcodes.

= Why an error message instead of a date? =
It's displayed in case start date begins after end date. To solve this please reset date.

= Why no start date in dashboard? =
Because I have added this feature in version 4.1.

All events posted before version 4.1 have 1 date only. But you can set a start and end date for each event afterwards.

= Why no event info, featured image, event categories while adding an event? =
If these boxes are not present, they might be unchecked in Screen Options. 

= Why a 404 (nothing found) when I click the title link? =
This is mostly caused by a wrong permalink setting. Please reset it via Settings > Permalinks. 

= How do I list events in a template file? =
For upcoming events: `<?php echo do_shortcode( '[vsel]' ); ?>` 

For past events: `<?php echo do_shortcode( '[vsel-past-events]' ); ?>`

For current events: `<?php echo do_shortcode( '[vsel-current-events]' ); ?>`

For all events: `<?php echo do_shortcode( '[vsel-all-events]' ); ?>`

= What happens with my events when I uninstall VSEL? =
You can find more info about this at the Installation section.

= How can I make a donation? =
You like my plugin and you're willing to make a donation? Nice! There's a PayPal donate link on the WordPress plugin page and my website.

= Other question or comment? =
Please open a topic in plugin forum.


== Changelog ==
= Version 6.1 =
* added shortcode attribute to change default order of events (asc or desc)
* for more info please check readme file
* bugfix: changed time format in shortcode files

= Version 6.0 =
* you can now also list all events
* added file vsel-all
* changed time format in shortcode files: now time of your website is used
* updated files vsel and vsel-widget

= Version 5.9 =
* widget information field: save html is allowed
* added filter wp_kses_post
* updated file vsel-widget

= Version 5.8 =
* all php files: disable direct access to file

For all versions please check file changelog.


== Screenshots == 
1. Very Simple Event List (Twenty Sixteen theme).
2. Very Simple Event List single event (Twenty Sixteen theme).
3. Very Simple Event List widget (Twenty Sixteen theme).
4. Very Simple Event List (dashboard).
5. Very Simple Event List single event (dashboard).
6. Very Simple Event List widget (dashboard).
7. Very Simple Event List settingspage (dashboard).

changelog.txt
CHANGELOG
=========

= Version 6.1 =
* added shortcode attribute to change default order of events (asc or desc)
* for more info please check readme file
* bugfix: changed time format in shortcode files

= Version 6.0 =
* you can now also list all events
* added file vsel-all
* changed time format in shortcode files: now time of your website is used
* updated files vsel and vsel-widget

= Version 5.9 =
* widget information field: save html is allowed
* added filter wp_kses_post
* updated file vsel-widget

= Version 5.8 =
* all php files: disable direct access to file

= Version 5.7 =
* file vsel-single: forgot to change class .vsel-info into .vsel-image-info

= Version 5.6 =
* hide image or event info in widget, instead of both simultaneously
* changed class .vsel-info into .vsel-image-info
* added class .vsel-info to event info
* updated shortcode files
* updated file vsel-options

= Version 5.5 =
* file vsel-widget: fixed php7 error (thanks nx3d)

= Version 5.4 =
* file vsel-upcoming: forgot to support multi-day events as well (thanks Tom)
* changed class .hide into .vsel-hide to avoid conflict with css in other stylesheets

= Version 5.3 =
* files vsel-upcoming, vsel-past and vsel-current: fix in display order of excerpt and content

= Version 5.2 =
* bugfix in date display
* bugfix in single event template
* added feature: custom summary while adding an event
* added new event list: current events
* updated file uninstall
* updated readme file

= Version 5.1 =
* updated file vsel-widget: no shortcode needed to display your upcoming or past events
* widget now only accepts attributes
* for more info please check readme file
* updated files vsel-upcoming, vsel-past and vsel-single: error message in case of wrong date

= Version 5.0 =
* updated files vsel and vsel-widget
* updated js file vsel-datepicker
* updated css file vsel-datepicker
* removed localization from datepicker (supported in WordPress 4.6)
* localized all placeholders
* added link to settingspage
* changed classes vsel-meta-text and vsel-all-events into vsel-widget-text and vsel-widget-link

= Version 4.9 =
* readme file: bunch of textual changes

= Version 4.8 =
* added file changelog
* updated readme file

= Version 4.7 =
* updated files vsel-widget and style
* widget: set a link to all events
* updated text strings
* updated readme file
 
= Version 4.6 =
* bugfix: option to hide date in single event

= Version 4.5 =
* added feature to link title to single event
* added the read more link on single event list too
* updated settingspage
* updated readme file

= Version 4.4 =
* relocated option (hide date) from event form to settingspage 
* updated settingspage
* changed file names
* updated readme file
* several small adjustments

= Version 4.3 =
* settingspage: Show a summary (excerpt) instead of all content in event list
* updated readme file

= Version 4.2 =
* finetuned the multi-day event display in frontend
* updated readme file

= Version 4.1 =
* added file vsel_single: VSEL now supports single events
* added multi-day feature with start end end date
* both shortcode files: hide pagination in widget because it's not working properly
* updated readme file
 
= Version 4.0 =
* bugfix: changed US dateformat in datepicker back to year-month-day

= Version 3.9 =
* added checkbox to open link in new window or the same window
* changed US dateformat in datepicker from year-month-day into month-day-year
* both shortcode files: wrapped next/prev link in div instead of paragraph tag 
* updated file uninstall

= Version 3.8 =
* file vsel: date column in dashboard is sortable now
* both shortcode files: wrapped next/prev link in paragraph tag
* updated file vsel_style

= Version 3.7 =
* file vsel: changed filter name
* updated text strings
* updated readme file

= Version 3.6 =
* important: changed past events shortcode into vsel-past-events
* updated readme file

= Version 3.5 =
* added shortcode attributes for event date, event time and event location
* updated settings to hide elements in widget event list (thanks Danny)
* added argument in query to avoid issue with very old events (thanks hallvors)
* updated text strings
* updated readme file
 
= Version 3.4 =
* added widget functionality
* options page: added settings to hide elements in widget event list
* updated file uninstall (thanks Danny)
* updated file vsel_style
* removed info about add-on (premium) version
* updated readme file

= Version 3.3 =
* added options page 
* added checkbox to keep all events and settings when uninstalling plugin
* added info about add-on (premium) version
* updated readme file

= Version 3.2 =
* request and fix: changed alignment of next and prev link

= Version 3.1 =
* localized the datepicker
* localized dateformat: European and US dateformat
* request: added checkbox to hide date in frontend
* added a PayPal donate link
* updated readme file

= Version 3.0 =
* relocated file vsel_style to folder css
* updated file vsel
* updated readme file

= Version 2.9 =
* file style: bugfix, in some cases event was not displayed full width (thanks Emmy)
* file style: added class for event title

= Version 2.8 =
* request: added shortcode attribute to set number of event on page
* request: added class for each meta-tag paragraph
* updated file vsel_style
* updated readme file

= Version 2.7 =
* removed translations: plugin now supports WordPress language packs
* added event categories: display events from certain categories using a shortcode attribute, more info about this at the Installation section 
* added custom URL link label to replace the static 'More info'
* updated readme file

= Version 2.6 =
* added file uninstall.php so settings in database (including all events) are removed when uninstalling plugin
* datepicker update

= Version 2.5 =
* updated Data Validation and Escaping
* fixed mistake in French translation (thanks Sylva) 

= Version 2.4 =
* fix: in previous versions content was displayed without linebreaks (thanks Jesper) 


vsel-widget.php
<?php

// disable direct access
if ( ! defined( 'ABSPATH' ) ) {
    exit;
}

class vsel_widget extends WP_Widget {
	// Constructor 
	public function __construct() {
		$widget_ops = array( 'classname' => 'vsel_widget', 'description' => __('Display your events in a widget.', 'very-simple-event-list') );
		parent::__construct( 'vsel_widget', __('Very Simple Event List', 'very-simple-event-list'), $widget_ops );
	}

	// Set widget and title in dashboard
	function form( $instance ) {
		$instance = wp_parse_args( $instance, array(
			'vsel_title' => '',
			'vsel_text' => '',
			'vsel_shortcode' => '',
			'vsel_attributes' => '',
			'vsel_all_events_link' => '',
			'vsel_all_events_label' => ''
		));
		$vsel_title = !empty( $instance['vsel_title'] ) ? $instance['vsel_title'] : __('Very Simple Event List', 'very-simple-event-list'); 
		$vsel_text = $instance['vsel_text'];
		$vsel_shortcode = $instance['vsel_shortcode'];
		$vsel_attributes = $instance['vsel_attributes'];
		$vsel_all_events_link = $instance['vsel_all_events_link'];
		$vsel_all_events_label = $instance['vsel_all_events_label'];

		?> 
		<p> 
		<label for="<?php echo $this->get_field_id( 'vsel_title' ); ?>"><?php _e('Title', 'very-simple-event-list'); ?>:</label> 
		<input class="widefat" id="<?php echo $this->get_field_id( 'vsel_title' ); ?>" name="<?php echo $this->get_field_name( 'vsel_title' ); ?>" type="text" maxlength='50' value="<?php echo esc_attr( $vsel_title ); ?>">
 		</p> 
		<p>
		<label for="<?php echo $this->get_field_id('vsel_text'); ?>"><?php _e('Information', 'very-simple-event-list'); ?>:</label>
		<textarea class="widefat monospace" rows="6" cols="20" id="<?php echo $this->get_field_id('vsel_text'); ?>" name="<?php echo $this->get_field_name('vsel_text'); ?>"><?php echo wp_kses_post( $vsel_text ); ?></textarea>
		</p>
		<p> 
		<label for="<?php echo $this->get_field_id( 'vsel_shortcode' ); ?>"><?php _e( 'List', 'very-simple-event-list' ); ?>:</label> 
		<select class="widefat" id="<?php echo $this->get_field_id( 'vsel_shortcode' ); ?>" name="<?php echo $this->get_field_name( 'vsel_shortcode' ); ?>";"> 
			<option value='upcoming'<?php echo ($vsel_shortcode == 'upcoming')?'selected':''; ?>><?php _e( 'Upcoming events', 'very-simple-event-list' ); ?></option> 
			<option value='past'<?php echo ($vsel_shortcode == 'past')?'selected':''; ?>><?php _e( 'Past events', 'very-simple-event-list' ); ?></option> 
			<option value='current'<?php echo ($vsel_shortcode == 'current')?'selected':''; ?>><?php _e( 'Current events', 'very-simple-event-list' ); ?></option> 
			<option value='all'<?php echo ($vsel_shortcode == 'all')?'selected':''; ?>><?php _e( 'All events', 'very-simple-event-list' ); ?></option> 
		</select> 
		</p>
		<p> 
		<label for="<?php echo $this->get_field_id( 'vsel_attributes' ); ?>"><?php _e('Attributes', 'very-simple-event-list'); ?>:</label> 
		<input class="widefat" id="<?php echo $this->get_field_id( 'vsel_attributes' ); ?>" name="<?php echo $this->get_field_name( 'vsel_attributes' ); ?>" type="text" maxlength='200' placeholder="<?php _e( 'Example: posts_per_page=2', 'very-simple-event-list' ); ?>" value="<?php echo esc_attr( $vsel_attributes ); ?>">
 		</p> 
		<p> 
		<label for="<?php echo $this->get_field_id( 'vsel_all_events_link' ); ?>"><?php _e('Link to all events', 'very-simple-event-list'); ?>:</label> 
		<input class="widefat" id="<?php echo $this->get_field_id( 'vsel_all_events_link' ); ?>" name="<?php echo $this->get_field_name( 'vsel_all_events_link' ); ?>" type="text" maxlength='150' placeholder="<?php _e( 'Example: mydomain.com/events', 'very-simple-event-list' ); ?>" value="<?php echo esc_url( $vsel_all_events_link ); ?>">
 		</p> 
		<p> 
		<label for="<?php echo $this->get_field_id( 'vsel_all_events_label' ); ?>"><?php _e('Link label', 'very-simple-event-list'); ?>:</label> 
		<input class="widefat" id="<?php echo $this->get_field_id( 'vsel_all_events_label' ); ?>" name="<?php echo $this->get_field_name( 'vsel_all_events_label' ); ?>" type="text" maxlength='100' placeholder="<?php _e( 'Example: All events', 'very-simple-event-list' ); ?>" value="<?php echo esc_attr( $vsel_all_events_label ); ?>">
 		</p> 
		<p><?php _e( 'Info about attributes', 'very-simple-event-list' ); ?>: <a href="https://wordpress.org/plugins/very-simple-event-list/installation" target="_blank"><?php _e( 'Installation', 'very-simple-event-list' ); ?></a></p>
		<p><?php _e( 'Hide elements in widget', 'very-simple-event-list' ); ?>: <a href="<?php echo admin_url( 'options-general.php?page=vsel' ); ?>"><?php _e( 'Settings', 'very-simple-event-list' ); ?></a></p>
		<?php 
	}

	// Update widget 
	function update( $new_instance, $old_instance ) {
		$instance = $old_instance;

		// Sanitize content
		$instance['vsel_title'] = strip_tags( $new_instance['vsel_title'] );
		$instance['vsel_text'] = wp_kses_post( $new_instance['vsel_text'] );
		$instance['vsel_shortcode'] = strip_tags( $new_instance['vsel_shortcode'] );
		$instance['vsel_attributes'] = strip_tags( $new_instance['vsel_attributes'] );
		$instance['vsel_all_events_link'] = esc_url_raw( $new_instance['vsel_all_events_link'] );
		$instance['vsel_all_events_label'] = strip_tags( $new_instance['vsel_all_events_label'] );

		return $instance;
	}

	// Display widget with event list in frontend 
	function widget( $args, $instance ) {
		if ( empty( $instance['vsel_all_events_label'] ) ) { 
			$instance['vsel_all_events_label'] = esc_attr__( 'All events', 'very-simple-event-list' );
		}

		echo $args['before_widget']; 

		if ( !empty( $instance['vsel_title'] ) ) { 
			echo $args['before_title'] . apply_filters( 'widget_title', $instance['vsel_title'] ). $args['after_title']; 
		} 

		if ( !empty( $instance['vsel_text'] ) ) { 
			echo '<div class="vsel-widget-text">';
			echo wpautop( wp_kses_post($instance['vsel_text']) );
			echo '</div>';
		}
		if ( $instance['vsel_shortcode'] == 'past' ) { 
			$content = '[vsel-past-events ';
		} else if ( $instance['vsel_shortcode'] == 'current' ) { 
			$content = '[vsel-current-events ';
		} else if ( $instance['vsel_shortcode'] == 'all' ) { 
			$content = '[vsel-all-events ';
		} else {
			$content = '[vsel ';
		}
		if ( !empty( $instance['vsel_attributes'] ) ) { 
			$content .= $instance['vsel_attributes'];
		}
		$content .= ']';
		echo do_shortcode( $content );

		if ( !empty( $instance['vsel_all_events_link'] ) ) { 
			echo '<div class="vsel-widget-link">' . sprintf( '<a href="%1$s">%2$s</a>', esc_url($instance['vsel_all_events_link']), esc_attr($instance['vsel_all_events_label']) ) . '</div>';
		} 

		echo $args['after_widget']; 
	}
}

?>

vsel.php
<?php
/*
 * Plugin Name: Very Simple Event List
 * Description: This is a very simple plugin to display a list of events. Use a shortcode to display events on a page or use the widget. For more info please check readme file.
 * Version: 6.1
 * Author: Guido van der Leest
 * Author URI: http://www.guidovanderleest.nl
 * License: GNU General Public License v3 or later
 * License URI: http://www.gnu.org/licenses/gpl-3.0.html
 * Text Domain: very-simple-event-list
 * Domain Path: /translation
 */

// disable direct access
if ( ! defined( 'ABSPATH' ) ) {
    exit;
}


// load plugin text domain
function vsel_init() { 
	load_plugin_textdomain( 'very-simple-event-list', false, dirname( plugin_basename( __FILE__ ) ) . '/translation' );
}
add_action('plugins_loaded', 'vsel_init');


// enqueue css script
function vsel_frontend_scripts() {	
	if(!is_admin())	{
		wp_enqueue_style( 'vsel_style', plugins_url('/css/vsel-style.css',__FILE__ ) );
	}
}
add_action('wp_enqueue_scripts', 'vsel_frontend_scripts');


// the sidebar widget
function register_vsel_widget() {
	register_widget( 'vsel_widget' );
}
add_action( 'widgets_init', 'register_vsel_widget' );


// set date formats for datepicker
function vsel_datepicker_dateformat( $dateformat ) { 
	if ($dateformat == 'j F Y' || $dateformat == 'd/m/Y' || $dateformat == 'd-m-Y') {
		$dateformat = 'dd-mm-yy'; 
	} else {
		$dateformat = 'yy-mm-dd'; 
	}
	return $dateformat; 
}


// enqueue datepicker script
function vsel_enqueue_date_picker(){ 
	global $wp_locale; 
	global $post_type; 

	if( 'event' != $post_type ) 
	return;
	wp_enqueue_script( 'vsel_datepicker_script', plugins_url( '/js/vsel-datepicker.js' , __FILE__ ), array('jquery', 'jquery-ui-datepicker') ); 
	wp_enqueue_style( 'vsel_datepicker_style', plugins_url( '/css/vsel-datepicker.css',__FILE__ ) );

	// datepicker args
	$vsel_datepicker_args = array(
		'dateFormat' => vsel_datepicker_dateformat( get_option( 'date_format' ) )
	);

	// localize script with data for datepicker
	wp_localize_script( 'vsel_datepicker_script', 'objectL10n', $vsel_datepicker_args );
}
add_action( 'admin_enqueue_scripts', 'vsel_enqueue_date_picker' ); 


// create eventspage in dashboard
function vsel_custom_postype() { 
	$vsel_labels = array( 
		'name' => __( 'Events', 'very-simple-event-list' ), 
		'all_items' => __( 'All Events', 'very-simple-event-list' ), 
		'singular_name' => __( 'Event', 'very-simple-event-list' ), 
		'add_new' => __( 'Add New', 'very-simple-event-list' ), 
		'add_new_item' => __( 'Add New Event', 'very-simple-event-list' ), 
		'edit_item' => __( 'Edit Event', 'very-simple-event-list' ), 
		'new_item' => __( 'New Event', 'very-simple-event-list' ), 
		'view_item' => __( 'View Event', 'very-simple-event-list' ), 
		'search_items' => __( 'Search Events', 'very-simple-event-list' ), 
		'not_found' => __( 'No events found', 'very-simple-event-list' ), 
		'not_found_in_trash' => __( 'No events found in Trash', 'very-simple-event-list' ), 
	); 
	$vsel_args = array( 
		'label' => __( 'Events', 'very-simple-event-list' ), 
		'labels' => $vsel_labels, 
		'public' => true, 
		'can_export' => true, 
		'show_in_nav_menus' => false, 
		'show_ui' => true, 
		'capability_type' => 'post', 
		'taxonomies' => array('event_cat'),
 		'supports'=> array('title', 'thumbnail', 'editor'), 
	); 
	register_post_type( 'event', $vsel_args); 
}
add_action( 'init', 'vsel_custom_postype' ); 


// create event categories
function vsel_taxonomy() { 
	register_taxonomy( 'event_cat', 'event', array( 'label' => __( 'Event Categories', 'very-simple-event-list' ), 'hierarchical' => true, ) ); 
}
add_action( 'init', 'vsel_taxonomy' ); 


// create metabox
function vsel_metabox() { 
	add_meta_box( 
		'vsel-event-metabox', 
		__( 'Event Info', 'very-simple-event-list' ), 
		'vsel_metabox_callback', 
		'event', 
		'side', 
		'default' 
	); 
} 
add_action( 'add_meta_boxes', 'vsel_metabox' );


function vsel_metabox_callback( $post ) { 
	// generate a nonce field 
	wp_nonce_field( 'vsel_meta_box', 'vsel_nonce' ); 
	
	// get previously saved meta values (if any) 
	$event_start_date = get_post_meta( $post->ID, 'event-start-date', true );
	$event_date = get_post_meta( $post->ID, 'event-date', true );
	$event_time = get_post_meta( $post->ID, 'event-time', true ); 
	$event_location = get_post_meta( $post->ID, 'event-location', true ); 
	$event_link = get_post_meta( $post->ID, 'event-link', true ); 
	$event_link_label = get_post_meta( $post->ID, 'event-link-label', true ); 
	$event_link_target = get_post_meta( $post->ID, 'event-link-target', true ); 
	$event_summary = get_post_meta( $post->ID, 'event-summary', true ); 

	// get date if saved else set it to current date 
	$event_start_date = !empty( $event_start_date ) ? $event_start_date : time(); 
	$event_date = !empty( $event_date ) ? $event_date : time(); 

	// set dateformat to match datepicker 
	$dateformat = get_option( 'date_format' );
	if ($dateformat == 'j F Y' || $dateformat == 'd/m/Y' || $dateformat == 'd-m-Y') {
		$dateformat = 'd-m-Y'; 
	} else {
		$dateformat = 'Y-m-d'; 
	}

	// metabox fields
	?> 
	<p><label for="vsel-start-date"><?php _e( 'Start date', 'very-simple-event-list' ); ?></label> 
	<input class="widefat" id="vsel-start-date" type="text" name="vsel-start-date" required maxlength="10" placeholder="<?php _e( 'Use datepicker', 'very-simple-event-list' ); ?>" value="<?php echo date_i18n( $dateformat, esc_attr( $event_start_date ) ); ?>" /></p>
	<p><label for="vsel-date"><?php _e( 'End date', 'very-simple-event-list' ); ?></label> 
	<input class="widefat" id="vsel-date" type="text" name="vsel-date" required maxlength="10" placeholder="<?php _e( 'Use datepicker', 'very-simple-event-list' ); ?>" value="<?php echo date_i18n( $dateformat, esc_attr( $event_date ) ); ?>" /></p>
	<p><label for="vsel-time"><?php _e( 'Time', 'very-simple-event-list' ); ?></label> 
	<input class="widefat" id="vsel-time" type="text" name="vsel-time" maxlength="100" placeholder="<?php _e( 'Example: 16.00 - 18.00', 'very-simple-event-list' ); ?>" value="<?php echo esc_attr( $event_time ); ?>" /></p>
	<p><label for="vsel-location"><?php _e( 'Location', 'very-simple-event-list' ); ?></label> 
	<input class="widefat" id="vsel-location" type="text" name="vsel-location" maxlength="100" placeholder="<?php _e( 'Example: Times Square', 'very-simple-event-list' ); ?>" value="<?php echo esc_attr( $event_location ); ?>" /></p>
	<p><label for="vsel-link"><?php _e( 'Link', 'very-simple-event-list' ); ?></label> 
	<input class="widefat" id="vsel-link" type="text" name="vsel-link" maxlength="150" placeholder="<?php _e( 'Example: wordpress.org', 'very-simple-event-list' ); ?>" value="<?php echo esc_url( $event_link ); ?>" /></p>
	<p><label for="vsel-link-label"><?php _e( 'Link label', 'very-simple-event-list' ); ?></label> 
	<input class="widefat" id="vsel-link-label" type="text" name="vsel-link-label" maxlength="100" placeholder="<?php _e( 'Example: More info', 'very-simple-event-list' ); ?>" value="<?php echo esc_attr( $event_link_label ); ?>" /></p>
	<p><input class="checkbox" id="vsel-link-target" type="checkbox" name="vsel-link-target" value="yes" <?php checked( $event_link_target, 'yes' ); ?> /> 
	<label for="vsel-link-target"><?php _e('Open link in new window', 'very-simple-event-list'); ?></label></p>
	<p><label for="vsel-summary"><?php _e( 'Custom summary', 'very-simple-event-list' ); ?></label> 
	<textarea id="vsel-summary" name="vsel-summary" class="large-text" rows="6" maxlength="150" placeholder="<?php _e( 'This will replace the default summary', 'very-simple-event-list' ); ?>"><?php echo esc_textarea( $event_summary); ?></textarea></p>
	<?php 
}


// save event
function vsel_save_event_info( $post_id ) { 
	// check if nonce is set
	if ( ! isset( $_POST['vsel_nonce'] ) ) {
		return;
	}
	// verify that nonce is valid
	if ( ! wp_verify_nonce( $_POST['vsel_nonce'], 'vsel_meta_box' ) ) {
		return;
	}
	// if this is an autosave, our form has not been submitted, so do nothing
	if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
		return;
	}
	// check user permissions
	if ( isset( $_POST['post_type'] ) && 'page' == $_POST['post_type'] ) {
		if ( ! current_user_can( 'edit_page', $post_id ) ) {
			return;
		}
	} else {
		if ( ! current_user_can( 'edit_post', $post_id ) ) {
			return;
		}
	}
	// checking values and save fields 
	if ( isset( $_POST['vsel-start-date'] ) ) { 
		update_post_meta( $post_id, 'event-start-date', sanitize_text_field(strtotime( $_POST['vsel-start-date'] ) ) ); 
	} 
	if ( isset( $_POST['vsel-date'] ) ) { 
		update_post_meta( $post_id, 'event-date', sanitize_text_field(strtotime( $_POST['vsel-date'] ) ) ); 
	} 
	if ( isset( $_POST['vsel-time'] ) ) { 
		update_post_meta( $post_id, 'event-time', sanitize_text_field( $_POST['vsel-time'] ) ); 
	} 
	if ( isset( $_POST['vsel-location'] ) ) { 
		update_post_meta( $post_id, 'event-location', sanitize_text_field( $_POST['vsel-location'] ) ); 
	} 
	if ( isset( $_POST['vsel-link'] ) ) { 
		update_post_meta( $post_id, 'event-link', esc_url_raw( $_POST['vsel-link'] ) ); 
	} 
	if ( isset( $_POST['vsel-link-label'] ) ) { 
		update_post_meta( $post_id, 'event-link-label', sanitize_text_field( $_POST['vsel-link-label'] ) ); 
	} 
	if ( isset( $_POST['vsel-link-target'] ) ) { 
		update_post_meta( $post_id, 'event-link-target', 'yes' ); 
	} else {
		update_post_meta( $post_id, 'event-link-target', 'no' ); 
	} 
	if ( isset( $_POST['vsel-summary'] ) ) { 
		update_post_meta( $post_id, 'event-summary', esc_textarea( $_POST['vsel-summary'] ) ); 
	} 
} 
add_action( 'save_post', 'vsel_save_event_info' );


// dashboard event columns
function vsel_custom_columns( $defaults ) { 
	unset( $defaults['date'] );
	$defaults['event_start_date'] = __( 'Start date', 'very-simple-event-list' ); 
	$defaults['event_date'] = __( 'End date', 'very-simple-event-list' ); 
	$defaults['event_time'] = __( 'Time', 'very-simple-event-list' ); 
	$defaults['event_location'] = __( 'Location', 'very-simple-event-list' ); 
	return $defaults; 
} 
add_filter( 'manage_event_posts_columns', 'vsel_custom_columns', 10 );

function vsel_custom_columns_content( $column_name, $post_id ) { 
	if ( 'event_start_date' == $column_name ) { 
		$start_date = get_post_meta( $post_id, 'event-start-date', true ); 
		if(!empty( $start_date ) ) { 
			echo date_i18n( get_option( 'date_format' ), $start_date ); 
		}
	} 
	if ( 'event_date' == $column_name ) { 
		$date = get_post_meta( $post_id, 'event-date', true ); 
		if(!empty( $date ) ) { 
			echo date_i18n( get_option( 'date_format' ), $date ); 
		}
	} 
	if ( 'event_time' == $column_name ) { 
		$time = get_post_meta( $post_id, 'event-time', true ); 
		echo $time; 
	} 
	if ( 'event_location' == $column_name ) { 
		$location = get_post_meta( $post_id, 'event-location', true ); 
		echo $location; 
	} 
} 
add_action( 'manage_event_posts_custom_column', 'vsel_custom_columns_content', 10, 2 );


// make event date column sortable
function vsel_column_register_sortable( $columns ) {
	$columns['event_start_date'] = 'event-start-date';
	$columns['event_date'] = 'event-date';
	return $columns;
}
add_filter( 'manage_edit-event_sortable_columns', 'vsel_column_register_sortable' );

function vsel_start_date_column_orderby( $vars ) {
	if(is_admin()) {
		if ( isset( $vars['orderby'] ) && 'event-start-date' == $vars['orderby'] ) {
			$vars = array_merge( $vars, array(
				'meta_key' => 'event-start-date',
				'orderby' => 'meta_value_num'
			) );
		}
	}
	return $vars;
}
add_filter( 'request', 'vsel_start_date_column_orderby' );

function vsel_date_column_orderby( $vars ) {
	if(is_admin()) {
		if ( isset( $vars['orderby'] ) && 'event-date' == $vars['orderby'] ) {
			$vars = array_merge( $vars, array(
				'meta_key' => 'event-date',
				'orderby' => 'meta_value_num'
			) );
		}
	}
	return $vars;
}
add_filter( 'request', 'vsel_date_column_orderby' );


// add class to pagination
function vsel_prev_posts() { 
	return 'class="prev"'; 
} 
add_filter('previous_posts_link_attributes', 'vsel_prev_posts', 10); 

function vsel_next_posts() { 
	return 'class="next"'; 
}
add_filter('next_posts_link_attributes', 'vsel_next_posts', 10); 


// add settings link
function vsel_action_links ( $links ) { 
	$settingslink = array( '<a href="'. admin_url( 'options-general.php?page=vsel' ) .'">'. __('Settings', 'very-simple-event-list') .'</a>', ); 
	return array_merge( $links, $settingslink ); 
} 
add_filter( 'plugin_action_links_' . plugin_basename(__FILE__), 'vsel_action_links' ); 


// include files
include 'vsel-upcoming.php';
include 'vsel-current.php';
include 'vsel-past.php';
include 'vsel-all.php';
include 'vsel-widget.php';
include 'vsel-options.php';
include 'vsel-single.php';

?>